### ISSUE TITLE
[BUG] Add-PnPListItemAttachment: The requested operation requires an element of type 'Object', but the target element has type 'String'.

### ISSUE THREAD

**Issue by LastGunslinger** [pnp/powershell#2445]
### Reporting an Issue or Missing Feature
Issue

### Expected behavior 
Using the Add-PnPListItemAttachment cmdlet should upload a file from the local file system as an attachment to a SharePoint list item.

### Actual behavior 
Command fails with error "Add-PnPListItemAttachment: The requested operation requires an element of type 'Object', but the target element has type 'String'."

### Steps to reproduce behavior
`Add-PnPListItemAttachment -List 'My List Name' -Identity 1 -Path './path/to/file'`

I have confirmed beforehand that I am connected to the correct site, that the list name is correct(I can use the exact same name with the Get-PnPList command and it is returned as expected), and that the item ID listed as the identity exists.

### What is the version of the Cmdlet module you are running?
1.11.0

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows
- [ ] Linux
- [x] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
hi @LastGunslinger , unable to reproduce this issue.

![image](https://user-images.githubusercontent.com/8832018/194941440-824d1e89-1d38-4e1f-b966-10287e45c501.png)

Are you entering the correct path ? You can also specify the full path of the file.
If you are not specifying full path, it will take the current session path and try to find the file there, maybe that's not working ?

**Comment by LastGunslinger** [comment]
@gautamdsheth, I found a clue as to what may be happening, but I have no idea why this would be the case. I entered this in the console to see if perhaps I was using an incorrect parameter set:

`Get-Help Add-PnPListItemAttachment -Detailed`

What I got returned was this:

> PS /Users/rjoh716/projects/powershell> Get-Help Add-PnPListItemAttachment -Detailed
> 
> NAME
>     Add-PnPListItemAttachment
>     
> SYNOPSIS
>     Removes attachment from the specified list item in the SharePoint list
>     
>     
> SYNTAX
>     Add-PnPListItemAttachment [-List] <ListPipeBind> -FileName <String> -Recycle <String> [-Force <String>] -Identity <ListItemPipeBind> [-Connection <PnPConnection>] [<CommonParameters>]
>     
>     Add-PnPListItemAttachment [-List] <ListPipeBind> -All -Recycle <String> [-Force <String>] -Identity <ListItemPipeBind> [-Connection <PnPConnection>] [<CommonParameters>]
>     
>     
> DESCRIPTION
>     
>     
> 
> PARAMETERS
>     -FileName <String>
>         Specify name of the attachment to delete from list item. The filename is not case sensitive.
>         
>     -All [<SwitchParameter>]
>         Specify if you want to delete or recycle all the list item attachments.
>         
>     -Recycle <String>
>         Specify if you want to send the attachment(s) to the recycle bin.
>         
>     -Force <String>
>         Specifying the Force parameter will skip the confirmation question.
>         
>     -List <ListPipeBind>
>         The ID, Title or Url of the list. Note that when providing the name of the list, the list name is case-sensitive.
>         
>     -Identity <ListItemPipeBind>
>         The ID of the listitem, or actual ListItem object
>         
>     -Connection <PnPConnection>
>         Optional connection to be used by the cmdlet. Retrieve the value for this parameter by either specifying -ReturnConnection on Connect-PnPOnline or by executing Get-PnPConnection.
>         
>     <CommonParameters>
>         This cmdlet supports the common parameters: Verbose, Debug,
>         ErrorAction, ErrorVariable, WarningAction, WarningVariable,
>         OutBuffer, PipelineVariable, and OutVariable. For more information, see
>         about_CommonParameters (https://go.microsoft.com/fwlink/?LinkID=113216). 
>     
>     -------------------------- EXAMPLE 1 --------------------------
>     
>     Remove-PnPListItemAttachment -List "Demo List" -Identity 1 -FileName test.txt
>     
>     Permanently delete an attachment from the list item with Id "1" in the "Demo List" SharePoint list with file name as test.txt.
>     -------------------------- EXAMPLE 2 --------------------------
>     
>     Remove-PnPListItemAttachment -List "Demo List" -Identity 1 -FileName test.txt -Recycle
>     
>     Removes an attachment from the list item with Id "1" in the "Demo List" SharePoint list with file name as test.txt and sends it to recycle bin.
>     -------------------------- EXAMPLE 3 --------------------------
>     
>     Remove-PnPListItemAttachment -List "Demo List" -Identity 1 -FileName test.txt -Recycle -Force
>     
>     Removes an attachment from the list item with Id "1" in the "Demo List" SharePoint list with file name as test.txt and sends it to recycle bin. It will not ask for confirmation from user.
>     -------------------------- EXAMPLE 4 --------------------------
>     
>     Remove-PnPListItemAttachment -List "Demo List" -Identity 1 -All -Recycle -Force
>     
>     Removes all attachments from the list item with Id "1" in the "Demo List" SharePoint list and sends them to recycle bin. It will not ask for confirmation from user.
>     -------------------------- EXAMPLE 5 --------------------------
>     
>     Remove-PnPListItemAttachment -List "Demo List" -Identity 1 -All
>     
>     Permanently deletes all attachments from the list item with Id "1" in the "Demo List" SharePoint list and sends them to recycle bin.
> REMARKS
>     To see the examples, type: "Get-Help Add-PnPListItemAttachment -Examples"
>     For more information, type: "Get-Help Add-PnPListItemAttachment -Detailed"
>     For technical information, type: "Get-Help Add-PnPListItemAttachment -Full"
>     For online help, type: "Get-Help Add-PnPListItemAttachment -Online"

This is obviously an error, why would I be getting the help details for the Remove-PnPListItemAttachment instead of Add-PnPListItemAttachment? I thought that maybe this was just an issue with my local machine, perhaps I somehow inadvertently aliased Remove- as Add- or something strange like that, so I had one of my coworkers do the same thing, and he also got the wrong help details. Is this some kind of clue that maybe the wrong command is being used, or is there a simpler explanation? I'm going to go over to my home machine(which doesn't have PnP.PowerShell installed) and install it fresh there to see if I get the same result.

**Comment by LastGunslinger** [comment]
@gautamdsheth, I just logged into my other machine and did a fresh install of PowerShell 7.2.6 and PnP.PowerShell 1.11.0, and this exact same thing is returned there. Using Get-Help for the Add-PnPListItemAttachment cmdlet returns the information for the Remove-PnPListItemAttachment cmdlet. I'm not sure if this is just an issue with the documentation, or with the module itself. Unfortunately I'm not able to try the file upload on my second computer since it is my personal computer and against our company policy.

**Comment by LastGunslinger** [comment]
Okay, last update. I think this specific error was due to an issue on the list itself. I tried to add an attachment to a different list, and it worked, then I tried on a different item on the same list, and it also worked. I'll work on my end to figure out where the issue lies here. I do believe that there may be an issue with Add-PnPListItemAttachment and the Get-Help cmdlet, but that seems to be a separate issue.
